app.js

```diff
@@ -1,44 +1,42 @@
 const express = require('express');
 const app = express();
-const {getTopics, getArticles, getArticleById, getArticleCommentById} = require('./controllers/con-topics')
+const {getTopics, getArticles, getArticleById, getArticleCommentById, postCommentByArticleId} = require('./controllers/con-topics')
 
+app.use(express.json());
 
 app.get('/api/topics', getTopics);
 app.get('/api/articles', getArticles)
 app.get('/api/articles/:article_id', getArticleById)
 app.get('/api/articles/:article_id/comments', getArticleCommentById)
+app.post('/api/articles/:article_id/comments', postCommentByArticleId)
+
+//handles all incorrect path errors
+app.all('*', (req, res, next) => {
+    res.status(404).send({msg: 'Path not found :( Try again.'})
+});
 
 
-//handles 400 errors (breaks sql rules so cannot exist)
 app.use((err, req, res, next) => {
-    if (err.code === '22P02') {
+    if (err.code === '22P02' || err.code === '23502') {
```

Comment:
I like the way you've added the extra error code here as an or to save having to reuse the code you've already written.

It would be good to move these error functions out to their own file now to keep the app.js clear.